cmd/gardener-extension-shoot-dns-service/app/app.go

```diff
@@ -61,19 +65,28 @@ func (o *Options) run(ctx context.Context) {
 		Burst: 130,
 	}, o.restOptions.Completed().Config)
 
-	mgr, err := manager.New(o.restOptions.Completed().Config, o.managerOptions.Completed().Options())
-	if err != nil {
-		controllercmd.LogErrAndExit(err, "Could not instantiate controller-manager")
-	}
+	mgrOpts := o.managerOptions.Completed().Options()
 
-	if err := dnsapi.AddToScheme(mgr.GetScheme()); err != nil {
+	if mgrOpts.Scheme == nil {
+		mgrOpts.Scheme = scheme.Scheme
```

Comment:
Passing `scheme.Scheme` directly to the manager can lead to race conditions.
Please create a new scheme via `runtime.NewScheme` and add the k8s scheme afterwards.

Reply:
Basically, I meant something like this:
```go
	mgrScheme := runtime.NewScheme()
	if err := scheme.AddToScheme(mgrScheme); err != nil {
		controllercmd.LogErrAndExit(err, "Could not update manager scheme (kubernetes)")
	}
	if err := dnsapi.AddToScheme(mgrScheme); err != nil {
		controllercmd.LogErrAndExit(err, "Could not update manager scheme (dnsapi)")
	}
	if err := extensionscontroller.AddToScheme(mgrScheme); err != nil {
		controllercmd.LogErrAndExit(err, "Could not update manager scheme")
	}
	mgrOpts.Scheme = mgrScheme
```